Addresses #4403, add Sql helper methods to retrieve U-factors, SHGC, or VT for glazing systems

joseph-robertson opened the pull request:
Pull request overview
---------------------

Tests demonstration https://github.com/NREL/OpenStudio/pull/4550

Questions:
- [ ] I'm creating an interior window, but it's not populating "Assembly xxx" fields in eplusout.sql:
![image](https://user-images.githubusercontent.com/8516790/158422882-c8c29b84-dfb3-44ac-a685-933dd8e1ed6a.png)

Link to the **Linux.deb** installer to use for CI Testing. If not set, it will default to latest official release.
[OpenStudio Installer]: http://openstudio-ci-builds.s3-website-us-west-2.amazonaws.com/PR-4550/OpenStudio-3.4.0-alpha%2B19ddb63e3e-Ubuntu-18.04.deb


This Pull Request is concerning:

 - [x] **Case 1 - `NewTest`:** a new test for a new model API class,

Depending on your answer, please fill out the required section below, and delete the three others.
Leave the review checklist in place.


----------------------------------------------------------------------------------------------------------

### Case 1: New test for a new model API class

Please include which class(es) you are adding a test to specifically test for.
Include a link to the OpenStudio Pull Request in which you are adding the new classes, or the class itself if already on develop.
<!---
> eg:
>
> This pull request is in relation with the Pull Request [NREL/OpenStudio#3031](https://github.com/NREL/OpenStudio/pull/3031), and  will specifically test for the following classes:
> * `AirTerminalSingleDuctConstantVolumeFourPipeBeam`
> * `CoilCoolingFourPipeBeam`
> * `CoilHeatingFourPipeBeam`
> * Additionally it explicitly tests for the existing class [TableMultiVariableLookUp](https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/model/TableMultiVariableLookup.hpp)
-->

#### Work Checklist

The following has been checked to ensure compliance with the guidelines:

 - [ ] Tests pass either:
     - [ ] with official OpenStudio release (include version):
         - [ ] A matching OSM test has been added from the successful run of the Ruby one with the official OpenStudio release
         - [ ] The label `AddedOSM` has been added to this PR
         - [ ] All new `out.osw` have been committed

     - [ ] with current develop (incude SHA):
         - [x] The label `PendingOSM` has been added to this PR
         - [x] A matching OSM test has not yet been added because the official release is pending, but `model_tests.rb` has a TODO.
            ```ruby
            def test_airterminal_cooledbeam_rb
              result = sim_test('airterminal_cooledbeam.rb')
            end

            # TODO: To be added in the next official release after: 2.5.0
            # def test_airterminal_fourpipebeam_osm
            #   result = sim_test('airterminal_fourpipebeam.osm')
            # end
            ```
        - [x] No `out.osw` have been committed as they need to be run with an official OpenStudio version


 - [ ] **Ruby test is stable**: when run multiple times on the same machine, it produces the same total site kBTU.
    Please paste the heatmap png generated after running the following commands:
     - [ ] I ensured that I assign systems/loads/etc in a repeatable manner (eg: if I assign stuff to thermalZones, I do `model.getThermalZones.sort_by{|z| z.name.to_s}.each do ...` so I am sure I put the same ZoneHVAC systems to the same zones regardless of their order)
     - [ ] I tested stability using `process_results.py` (see `python process_results.py --help` for usage).
    Please paste the text output or heatmap png generated after running the following commands:
        ```bash
        # Clean up all custom-tagged OSWs
        python process_results.py test-stability clean
        # Run your test 5 times in a row. Replace `testname_rb` (eg `airterminal_fourpipebeam_rb`)
        python process_results.py test-stability run -n testname_rb
        # Check that they all passed
        python process_results.py test-status --tagged
        # Check site kBTU differences
        python process_results.py heatmap --tagged

        ```

 - [ ] **Object has been added to `autosize_hvac.rb` to ensure the autosizedXXX values methods do work**

----------------------------------------------------------------------------------------------------------

### Review Checklist

 - [ ] Code style (indentation, variable names, strip trailing spaces)
 - [ ] Functional code review (it has to work!)
 - [ ] Matching OSM test has been added or `# TODO` added to `model_tests.rb`
 - [ ] Appropriate `out.osw` have been committed
 - [ ] Test is stable
 - [ ] Object is tested in `autosize_hvac` as appropriate
 - [ ] The appropriate labels have been added to this PR:
   - [ ] One of: `NewTest`, `TestFix`, `NewTestForExisting`, `Other`
   - [ ] If `NewTest`: add `PendingOSM` or `AddedOSM`

joseph-robertson committed:
Add new frame and divider test with nfrc setter.

joseph-robertson committed:
Add new call to model tests rb.

Commit:
Commit rubocop --auto-correct (Ruby ruby 2.7.5p203 (2021-11-24 revision f69aeb8314) [x86_64-linux], Rubocop 0.81.0)

joseph-robertson committed:
Try adding an interior window.

joseph-robertson committed:
Create zone on other side of interior window.

jmarrec commented on model/simulationtests/window_property_frame_and_divider.rb:
```diff
@@ -0,0 +1,98 @@
+  construction_set.setAdiabaticSurfaceConstruction(adiabatic_c)
+end
+# assert !adiabatic_surface.construction.empty?
+# assert adiabatic_surface.construction.get == adiabatic_c
```
Remove this, you're not testing adiabatic construction set here

joseph-robertson committed:
Remove creation of interior window.

jmarrec commented on model/simulationtests/window_property_frame_and_divider.rb:
```diff
@@ -0,0 +1,82 @@
+z = zones[0]
+z.setUseIdealAirLoads(true)
+
+# Get the first Outdoors surface, sorting by name to ensure consistency
+surfaces = model.getSurfaces.select { |s| s.outsideBoundaryCondition == 'Outdoors' }.sort_by { |s| s.name.to_s }
+
+adiabatic_surface = surfaces[0]
+# This will remove the subSurfaces (if any)
+adiabatic_surface.setOutsideBoundaryCondition('Adiabatic')
+# assert adiabatic_surface.construction.empty?
+
+# Only one here, but let's be safe
+construction_set = model.getDefaultConstructionSets.min_by { |d| d.name.to_s }
+# I know it's initialized, so I get it already
+ext_set = construction_set.defaultExteriorSurfaceConstructions.get
+# Same, I know it's initialized
+c = ext_set.wallConstruction.get
+adiabatic_c = c.clone(model).to_ConstructionBase.get
+adiabatic_c.setName('Adiabatic Construction')
+
+if Gem::Version.new(OpenStudio.openStudioVersion) > Gem::Version.new('2.7.1')
+  construction_set.setAdiabaticSurfaceConstruction(adiabatic_c)
+end
+# assert !adiabatic_surface.construction.empty?
+# assert adiabatic_surface.construction.get == adiabatic_c
```
```suggestion
```

jmarrec committed:
Merge branch 'develop' into issue-4403

jmarrec committed:
Remove copy paste from adiabatic_construction_set.rb

jmarrec committed:
Exercise **ALL** setters, so it serves as an API demonstration (which is a goal of this repo)

I also noticed an IDD issue: there is no IDD default for Frame Conductance but the E+ source code will read it as 0.0

jmarrec commented:
@joseph-robertson I rewrote the test, please take a look then merge away
